python-package/lightgbm/engine.py

```diff
@@ -370,6 +370,8 @@ def cv(params, train_set, num_boost_round=10,
 
     if metrics:
         params.setdefault('metric', [])
+        if isinstance(params['metric'], string_type):
```

Comment:
If just override params['metric'] with metrics,then we will lose the metrics set in the params.
If metrics is a set , params['metric'].extend(metrics)  can also work.